app/models/document_type.rb

```diff
@@ -12,7 +12,9 @@ def self.find(id)
   end
 
   def self.all
-    @all ||= begin
+    return @all if defined?(@all) && @all.present?
```

Comment:
Could you help me understand why this line is necessary? I was thinking the `||=` should still do the trick.

Reply:
You're quite right 👍 I originally set `@all = []` in the `self.clear` method, which made this line necessary. Then I followed your suggestion of setting it to `nil` and ended up building on it instead of removing it. Have now removed!